### DISCUSSION TITLE
Logout extensibility

### DISCUSSION THREAD

**Discussion by finkinfridom** [payloadcms/payload#1253] [Feature Requests & Ideas]
The `logout` button in the `<DefaultNav/>` component doesn't allow any kind of extensibility.

Let's say I'm integrating an SSO workflow where, especially at logout, the user must be redirected to the IdP logout page. At the moment there's no way to change the logout link nor force a `res.redirect` into the `hooks.afterLogout` method (as it throws an exception within expressjs).

**Comment by DanRibbens** [comment]
I agree there could be more extensibility on the frontend admin UI, but you probably don't need that for what you are trying to do. 

Take a look at custom endpoints documentation. https://payloadcms.com/docs/rest-api/overview#custom-endpoints

You can use that to extend your auth collections api routes. Add an endpoint using the path /users/logout that removes the cookie or whatever else your SSO logout needs are.

Does that answer your immediate needs?

**Reply by finkinfridom** [reply]
Unfortunately, during logout, the user's browser must redirect to the identity provider logout endpoint (so the third-parties cookies' got cleaned), and then it will be redirected to the payload UI.

Maybe, extending the payload's config with
```
components: {
  logout: []
}
```

could be an option.

**Reply by jmikrut** [reply]
Yep. I think this makes sense. Originally I was thinking the same way that @DanRibbens was, but I now see that this might not always be the way forward. @finkinfridom wanna take a crack at a PR for this? 

I wrote up a list of steps to follow for PRs here:
https://github.com/payloadcms/payload/discussions/1219

It might be helpful for you to look at. But overall I think the PR makes sense!

**Comment by finkinfridom** [comment]
I created a PR here: https://github.com/payloadcms/payload/pull/1274
